fastlane/lib/fastlane/actions/onesignal.rb

```diff
@@ -33,60 +44,69 @@ def self.run(params)
         payload["gcm_key"] = android_token unless android_token.nil?
         payload["android_gcm_sender_id"] = android_gcm_sender_id unless android_gcm_sender_id.nil?
 
-        # here's the actual lifting - POST to OneSignal
+        # here's the actual lifting - POST or PUT to OneSignal
 
         json_headers = { 'Content-Type' => 'application/json', 'Authorization' => "Basic #{auth_token}" }
-        uri = URI.parse('https://onesignal.com/api/v1/apps')
+        url = +'https://onesignal.com/api/v1/apps'
+        url << '/' + app_id if is_update
+        uri = URI.parse(url)
         http = Net::HTTP.new(uri.host, uri.port)
         http.use_ssl = true
-        response = http.post(uri.path, payload.to_json, json_headers)
+
+        if is_update
+          response = http.put(uri.path, payload.to_json, json_headers)
+        else
+          response = http.post(uri.path, payload.to_json, json_headers)
+        end
+
         response_body = JSON.parse(response.body)
 
         Actions.lane_context[SharedValues::ONE_SIGNAL_APP_ID] = response_body["id"]
         Actions.lane_context[SharedValues::ONE_SIGNAL_APP_AUTH_KEY] = response_body["basic_auth_key"]
 
-        check_response_code(response)
+        check_response_code(response, is_update)
       end
 
-      def self.check_response_code(response)
+      def self.check_response_code(response, is_update)
         case response.code.to_i
         when 200, 204
-          puts("Successfully created new OneSignal app".green)
+          puts("Successfully #{is_update ? 'updated' : 'created new'} OneSignal app".green)
         else
           UI.user_error!("Unexpected #{response.code} with response: #{response.body}")
         end
       end
 
       def self.description
-        "Create a new [OneSignal](https://onesignal.com/) application"
+        "Create or update a new [OneSignal](https://onesignal.com/) application"
       end
 
       def self.details
-        "You can use this action to automatically create a OneSignal application. You can also upload a `.p12` with password, a GCM key, or both."
+        "You can use this action to automatically create or update a OneSignal application. You can also upload a `.p12` with password, a GCM key, or both."
       end
 
       def self.available_options
         [
+          FastlaneCore::ConfigItem.new(key: :app_id,
+                                       env_name: "ONE_SIGNAL_APP_ID",
+                                       sensitive: true,
+                                       description: "OneSignal App ID",
```

Comment:
maybe add something here that it is to be used to update an app, and to `app_name` that it is meant to create a new one.

Reply:
Are the new texts clear enough?